src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -779,27 +760,30 @@ internal void Import()
                         continue;
                     }
 
-                    ICollection c = psObject.BaseObject as ICollection;
-                    if (c != null)
+                    if (psObject != null)
```

Comment:
Minor note: The loop has unnecessary nesting and duplication.  Consider handling psObject != null and c != null) first to reduce nesting.  Something like the following:

object result = _deserializer.Deserialize();
PSObject psObject = result as PSObject;

if (psObject != null)
{
    ICollection c = psObject.BaseObject as ICollection;
    if (c != null)
    {
        foreach (object o in c)
        {
            if (count >= first)
                break;

            if (skipped++ >= skip)
            {
                count++;
                _cmdlet.WriteObject(o);
            }
        }
        continue;
    }
}

if (skipped++ >= skip)
{
    count++;
    _cmdlet.WriteObject(result);
}